clientlib/src/main/java/com/yelp/nrtsearch/server/utils/ScriptParamsUtils.java

```diff
@@ -72,6 +72,21 @@ public static List<Object> decodeList(Script.ParamListValue listValue) {
     return nativeList;
   }
 
+  /**
+   * Decode a {@link com.yelp.nrtsearch.server.grpc.Script.ParamFloatVectorValue} message into an
+   * equivalent float array.
+   *
+   * @param vectorValue vector message
+   * @return float array
+   */
+  public static float[] decodeFloatVector(Script.ParamFloatVectorValue vectorValue) {
+    float[] vector = new float[vectorValue.getValuesCount()];
+    for (int i = 0; i < vectorValue.getValuesCount(); i++) {
+      vector[i] = vectorValue.getValues(i);
+    }
```

Comment:
Can we use the toArray method?

Reply:
I don't believe that will work. It would give you a `Float[]`, but we want to create the primitive version.